### PR TITLE
Gather Op

### LINKED ISSUE #1068: Update Gather operator

### PR DESCRIPTION
**Pull request by KateUnger**
**Description of changes:**

**Related Issues:**

Linked Issues:
- Issue #1068 

Issues closed by this PR:
- Closes #1068

- - -
This change is [Reviewable](https://reviewable.io/reviews/flexflow/FlexFlow/1165)

### CODE REVIEW

File: lib/kernels/src/cuda/gather_kernels.cu
Comment (reyna-abhyankar) [inline lib/kernels/src/cuda/gather_kernels.cu:1]: Based on the old kernel code, we should still template the gather index datatype.

File: lib/runtime/src/ops/gather.cc [hunk 1]
```diff
@@ -14,403 +14,181 @@
...
+  fwd.add_output_slot(OUTPUT);
+  fwd.add_weight_slot(INDEX);
+  register_task(GATHER_FWD_TASK_ID, "Gather Fwd", fwd, forward_task);
 }
 
-}; // namespace FlexFlow
+template <>
+void register_task<GATHER_BWD_TASK_ID>() {
+  OpTaskSignature bwd =
+      infer_bwd_signature(get_op_signature(GATHER_BWD_TASK_ID));
```
Comment (reyna-abhyankar) [inline lib/runtime/src/ops/gather.cc:189]:
```suggestion
      infer_bwd_signature(get_fwd_signature(GATHER_BWD_TASK_ID));
```
And I think fix the `register_fwd_task`

**Review by reyna-abhyankar — CHANGES_REQUESTED** [review]

**Comment by lockshaw** [comment]
@reyna-abhyankar Can this be closed now as it's superseded by #1389?

**Comment by lockshaw** [comment]
Superseded by #1389.

**Pull request closed without merging**